tests/records/test_dumpers.py

```diff
@@ -95,3 +96,181 @@ def test_esdumper_with_edtfext_parse_error(app, db, minimal_record):
+        extensions=[LocationsDumper()]
+    )
+
+    minimal_record['locations'] = {
+        'type': 'FeatureCollection',
+        'features': [{
+            'type': 'Feature',
+            'geometry': {
+                'type': 'Point',
+                'coordinates': [6.052778, 46.234167]
+            }
+        }]
+    }
+
+    record = BibliographicRecord.create(minimal_record)
+
+    # Dump it
+    dump = record.dumps(dumper=dumper)
+
+    # Types have been removed
+    assert 'type' not in dump['locations']
+    assert 'type' not in dump['locations']['features'][0]
+
+    # Centroid has been inferred
+    assert (
+        dump['locations']['features'][0]['centroid'] ==
+        minimal_record['locations']['features'][0]['geometry']['coordinates']
+    )
+
+    # And it round-trips
+    assert (
+        record.loads(dump, loader=dumper)['locations'] ==
+        minimal_record['locations']
+    )
+
+
+def test_locationsdumper_with_no_featurecollection(app, db, minimal_record):
+    dumper = ElasticsearchDumper(
+        extensions=[LocationsDumper()]
+    )
+
+    record = BibliographicRecord.create(minimal_record)
+
+    # Dump it
+    dump = record.dumps(dumper=dumper)
+
+    assert 'locations' not in dump
+    assert 'locations' not in record.loads(dump, loader=dumper)
+
+
+def test_locationsdumper_with_empty_featurecollection(app, db, minimal_record):
+    dumper = ElasticsearchDumper(
+        extensions=[LocationsDumper()]
+    )
+
+    minimal_record['locations'] = {
+        'type': 'FeatureCollection',
+        'features': [],
+    }
+
+    record = BibliographicRecord.create(minimal_record)
+
+    # Dump it
+    dump = record.dumps(dumper=dumper)
+
+    assert 'locations' in dump
+    assert 'type' not in dump['locations']
+    assert len(dump['locations']['features']) == 0
+
+    loaded = record.loads(dump, loader=dumper)
+    assert 'locations' in loaded
+    assert loaded['locations']['features'] == []
+
+
+@unittest.mock.patch('invenio_rdm_records.records.dumpers.shapely', None)
+def test_locationsdumper_with_polygon_and_no_shapely(app, db, minimal_record):
+    dumper = ElasticsearchDumper(
+        extensions=[LocationsDumper()]
+    )
+
+    minimal_record['locations'] = {
+        'type': 'FeatureCollection',
+        'features': [{
+            'type': 'Feature',
+            'geometry': {
+                'type': 'Polygon',
+                'coordinates': [
+                    [
+                        [100.0, 0.0], [101.0, 0.0], [101.0, 1.0], [100.0, 1.0],
+                        [100.0, 0.0],
+                    ]
+                ]
+            }
+        }],
+    }
+
+    record = BibliographicRecord.create(minimal_record)
+
+    with pytest.warns(UserWarning):
+        dump = record.dumps(dumper=dumper)
+
+    assert 'centroid' not in dump['locations']['features'][0]
+
+
+def test_locationsdumper_with_polygon_and_mock_shapely(
+    app, db, minimal_record
+):
+    with unittest.mock.patch(
```

Comment:
Nice! much clearer than the `@patch` decorator 🚀

Reply:
I tried the decorator first, and pytest complained about the argument not being a fixture :shrug:. But yep, the context manager is good :-)